presto-spi/src/main/java/com/facebook/presto/spi/plan/LimitNode.java

```diff
@@ -56,21 +56,30 @@ public LimitNode(
     @Override
     public List<PlanNode> getSources()
     {
-        return ImmutableList.of(source);
+        return singletonList(source);
     }
 
+    /**
+     * LimitNode only expects a single upstream PlanNode.
+     */
     @JsonProperty
     public PlanNode getSource()
     {
         return source;
     }
 
+    /**
+     * Get the limit `N` number of results to return.
+     */
     @JsonProperty
     public long getCount()
     {
         return count;
     }
 
+    /**
+     *  `LimitNode` only generates partial results of local worker when executed distributedly.
+     */
     @JsonProperty
     public boolean isPartial()
```

Comment:
Since we are moving this to SPI, is it worth discussing whether having an `isPartial` is better or switch the logic and have an `isFinal` is better? I feel that `isFinal` requires less explanation. Or maybe we should take the more generic approach as to define a `STEP` enum and use `PARTIAL` and `FINAL` explicitly.

Reply:
I would prefer `PARTIAL` + `FINAL` approach just like topN node

Reply:
I also believe having `PARTIAL` and `FINAL` steps be explicit is the easiest to understand.